### PR TITLE
WIP: Fix test bugs (and real bugs!)

### PR DESCRIPTION
**Pull request by jdfreder**
closes: #2

### CODE REVIEW

**Comment by minrk** [comment]
An improvement, but not quite passing.

**Comment by jdfreder** [comment]
No, I stopped part way through to pickup a friend from LAX.  I'll continue to work on this today.

**Comment by jdfreder** [comment]
Strange, looking at this closely, running some tests by hand, it looks like the entire state is being sent from the front-end to the back-end every time a value changes.  So the failure is legitimate, I'm not quite sure why it started failing now though...  Looking more

**Comment by jdfreder** [comment]
Spoke too soon, looks like that broke **all the things!**.

**Comment by jdfreder** [comment]
Python 2 problem should be fixed by https://github.com/ipython/ipython_kernel/pull/11 and my last commit here.

**Comment by jdfreder** [comment]
Success

**Comment by jdfreder** [comment]
@jasongrout or @SylvainCorlay would one of you like to review?

**Comment by SylvainCorlay** [comment]
This all looks good to me. For my own information, why do you need the conversion to string in widget.py?

**Comment by jdfreder** [comment]
import_item was getting Bytes or Unicode, which https://github.com/ipython/traitlets/blame/4b26dff292a23b89976afa7732894c5bb1969f03/traitlets/utils/importstring.py#L27 doesn't like in Python 2.x.  Python 3.4 was happy though.

**Comment by SylvainCorlay** [comment]
Ok, I checked out your branch and everything seems fine to me.

**Comment by jdfreder** [comment]
Do you want to click the green button?

File: ipython_widgets/widgets/widget.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def _call_widget_constructed(widget):
     @staticmethod
     def handle_comm_opened(comm, msg):
         """Static method, called when a widget is constructed."""
-        widget_class = import_item(msg['content']['data']['widget_class'])
+        widget_class = import_item(str(msg['content']['data']['widget_class']))
```
Comment (SylvainCorlay) [inline ipython_widgets/widgets/widget.py]: Note for self, this should not be needed.

**Pull request merged**